src/devices/machine/qvt102.h

```diff
@@ -0,0 +1,135 @@
+// license:BSD-3-Clause
+// copyright-holders:Robbbert, Dirk Best
+/****************************************************************************
+
+    Qume QVT-102/QVT-102A video terminal
+
+    Hardware:
+    - Motorola 6800 CPU
+    - Hitachi HD46505SP (Motorola 6845-compatible) CRTC
+    - Hitachi HD46850 (Motorola 6850-compatible) ACIA
+    - M58725P-15 (6116-compatible) (2k x 8bit RAM)
+    - Zilog Z8430 CTC
+    - 16.6698MHz Crystal
+    - 2x TC5514-APL + 3V battery, functioning as NVRAM
+
+    Keyboard: D8748D, 6.000MHz Crystal, Beeper
+
+    TODO:
+    - Support QVT-102A differences (bidirectional aux, different keyboard)
+    - Key click sounds weird
+
+****************************************************************************/
+
+#ifndef MAME_MACHINE_QVT102_H
+#define MAME_MACHINE_QVT102_H
+
+#pragma once
+
+#include "cpu/m6800/m6800.h"
+#include "cpu/mcs48/mcs48.h"
+#include "machine/input_merger.h"
+#include "machine/6850acia.h"
+#include "machine/z80ctc.h"
+#include "video/mc6845.h"
+#include "sound/spkrdev.h"
+#include "bus/rs232/rs232.h"
+#include "emupal.h"
+#include "screen.h"
+
+INPUT_PORTS_EXTERN(qvt102);
+
+class qvt102_device : public device_t
+{
+public:
+	qvt102_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+	// Interface to a RS232 connection.
+	auto rs232_conn_txd_handler() { return m_rs232_conn_txd_handler.bind(); }
+	auto rs232_conn_dtr_handler() { return m_rs232_conn_dtr_handler.bind(); }
+	auto rs232_conn_rts_handler() { return m_rs232_conn_rts_handler.bind(); }
+	DECLARE_WRITE_LINE_MEMBER(rs232_conn_dcd_w);
+	DECLARE_WRITE_LINE_MEMBER(rs232_conn_dsr_w);
+	DECLARE_WRITE_LINE_MEMBER(rs232_conn_ri_w);
+	DECLARE_WRITE_LINE_MEMBER(rs232_conn_cts_w);
+	DECLARE_WRITE_LINE_MEMBER(rs232_conn_rxd_w);
```

Comment:
Please clearly separate configuration member functions and live inputs.  Often configuration members are declared above the constructor to make things clearer.

Reply:
Could I leave these grouped together here just for a bit. I have in mind adding a new class defining these, so that the terminals have a common interface to the rs232 bus, and it will be quicker to clean up with them all together. This class might implement some common null modem options too, too refactor this code. Looking forward some terminals have more than one host interface, such as the tv900 and it would be nice to be able to route both to host rs232 ports - this area needs some 'architecture' design that might be better driven by other contributors.

Reply:
You should at least separate them and mark which ones are for configuration and which ones are inputs.  Also, the QVT-102 doesn't have an RI input as far as I can tell, and it has a clock output.